### PR TITLE
fix: deepspeed autotune user guide clarification

### PR DESCRIPTION
**Pull request by garrett361**
## Description

Simplifies the `DeepSpeedTrial` section of the user guide and also adds a warning about pipeline and model parallelism.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by ighodgao — APPROVED** [review]
Much clearer! Thank you

**Review by tayritenour — CHANGES_REQUESTED** [review]
@tara-det-ai should look at this

File: docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst [hunk 1]
```diff
@@ -301,3 +306,7 @@ semi-random search over the batch size.
 -  ``--early-stopping``: If provided, the experiment will terminate if a new best-configuration has
    not been found in the last ``early-stopping`` trials. Default: ``None``, corresponding to no such
    early stopping.
+
+.. warning::
+
+   This is a warning
```
[the remark this answers is not in the document]
Comment (garrett361) [reply docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst]: Done, oops.

**Comment by garrett361** [comment]
> @tara-det-ai should look at this

Yep, I think

File: docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst [hunk 2]
```diff
@@ -65,6 +65,12 @@ series of trials to search for the DS settings which optimize ``FLOPS_per_gpu``,
 ``custom`` experiment above. The search is informed both by the initial profiling trial and the
 results of each subsequent trial, all of whose results are fed back to the search runner.
 
+.. warning::
+
+   Determined's DeepSpeed Autotune is not currently compatible with pipeline or model parallelism.
+   The to-be-trained model must be a ``DeepSpeedEngine`` instance (not a ``PipelineEngine``
+   instance).
+
```
Comment (tara-det-ai) [inline docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst:73]: .. note::

This feature is not compatible with pipeline or model parallelism. The model to be trained must be a ``DeepSpeedEngine`` instance and not a ``PipelineEngine`` instance.

File: docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst [hunk 3]
```diff
@@ -78,31 +84,25 @@ will describe specific use cases and the changes needed for each.
...
 
-Determined's DeepSpeed Autotune works by inserting DeepSpeed configuration options into the
-``overwrite_deepspeed_args`` field of the ``hyperparameters`` dictionary which is seen by each
-trial. To take advantage of ``dsat``, you simply need to incorporate these "overwrite" values into
-your original configuration.
-
-.. note::
-
-   For more information about ``DeepSpeedTrial``, see :ref:`deepspeed-api`.
+In order to use Determined's DeepSpeed Autotune with ``DeepSpeedTrial`` there are two requirements.
```
Comment (tara-det-ai) [inline docs/model-dev-guide/apis-howto/deepspeed/autotuning.rst:87]: avoid references to specific counts. simplify "In order to" with "To".

To use Determine's DeepSpeed Autotune with ``DeepSpeedTrial``, you must meet the following requirements.

**Review by tara-det-ai — COMMENTED** [review]
requested changes

**Review by tara-det-ai — APPROVED** [review]
LGTM

**Review by tayritenour — APPROVED** [review]
LGTM

**Pull request merged**